portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -56,127 +64,232 @@ public void tearDown() {
-
-    when(postRequest.getParameter("text-input")).thenReturn("Test Comment #1");
+  public void testGetSingleComment() throws Exception {
+    long TIMESTAMP_ONE = 0;
+    Entity comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #1");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_ONE);
+    ds.put(comment);
 
     StringWriter stringWriter = new StringWriter();
     PrintWriter printWriter = new PrintWriter(stringWriter);
     when(getResponse.getWriter()).thenReturn(printWriter);
 
-    new DataServlet().doPost(postRequest, postResponse);
-    new DataServlet().doGet(getRequest, getResponse);
+    dataServelet.doPost(postRequest, postResponse);
+    dataServelet.doGet(getRequest, getResponse);
+
+    printWriter.flush(); //may not have flushed yet
+    String[] comments = convertStringToArray(stringWriter.toString());
+    assertThat(comments[0]).isEqualTo("Test Comment #1");
+  }
+  
+  @Test
+  public void testGetSomeComments() throws Exception {
+    int NUM_COMMENTS = 2;
+    long TIMESTAMP_ONE = 0;
+    Entity comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #1");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_ONE);
+    ds.put(comment);
+
+    long TIMESTAMP_TWO = 1;
+    comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #2");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_TWO);
+    ds.put(comment);
+
+    long TIMESTAMP_THREE = 2;
+    comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #3");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_THREE);
+    ds.put(comment);
+
+    long TIMESTAMP_FOUR = 3;
+    comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #4");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_FOUR);
+    ds.put(comment);
+
+    long TIMESTAMP_FIVE = 4;
+    comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #5");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_FIVE);
+    ds.put(comment);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(getResponse.getWriter()).thenReturn(printWriter);
+    when(getRequest.getParameter("max-num")).thenReturn(Integer.toString(NUM_COMMENTS));
+    dataServelet.doGet(getRequest, getResponse);
 
-    printWriter.flush(); //may not have flushed yets
-    System.out.println(stringWriter);
-    assertThat(stringWriter.toString()).contains("Test Comment #1");
+    printWriter.flush(); //may not have flushed yet
+    String[] comments = convertStringToArray(stringWriter.toString());
+    assertThat(comments[0]).isEqualTo("Test Comment #1");
+    assertThat(comments[1]).isEqualTo("Test Comment #2");  
+    assertThat(comments.length).isEqualTo(2);
```

Comment:
Have a gander at https://truth.dev/

Once comments is a `List`, these three assertions can be collapsed into one:

```java
assertThat(comments).containsExactly("Test Comment #1", "Test Comment #2");
```

Reply:
oh that's awesome! will update right now!